rfcs/0132-meson-builds-nix.md

```diff
@@ -0,0 +1,216 @@
+
+Albeit Ninja DSL is human-readable, it is not convenient to be manually written by human beings. As said before, Ninja is commonly used in tandem with other, higher-level build system in a two-pass fashion. In our present use case, the Meson interpreter converts Meson files to Ninja files that will be consumed by Ninja tool to effectively execute the building/deployment commands.
+
+## Design
+
+A carefully crafted set of Meson files should be included in order to describe how to deploy the Nix repository, generating all the expected artifacts (command line tools, libraries, configuration files, documentation etc.)
+
+This novel building infrastructure should be able to provide at least feature parity with the current quasi-autotools implementation, albeit in a different user interface.
+
+# Examples and Interactions
+[examples-and-interactions]: #examples-and-interactions
+
+## Example interaction
+
+Here is a table comparing some expected interactions:
+
+| Action    | Current                        | Meson+Ninja                          | Meson, backend-agnostic              |
+|-----------|--------------------------------|--------------------------------------|--------------------------------------|
+| Configure | `./configure --enable-gc=true` | `meson setup build_dir -Dgc=enabled` | `meson setup build_dir -Dgc=enabled` |
+| Build     | `make`                         | `ninja -C build_dir build`           | `meson -C build_dir compile`         |
+| Install   | `make install`                 | `ninja -C build_dir install`         | `meson -C build_dir install`         |
+| Uninstall | `make uninstall`               | `ninja -C build_dir unistall`        | `meson -C build_dir unistall`        |
+
+## Implementation
+
+Currently, @p01arst0rm is working on an implementation from scratch.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Below we will list some possible drawbacks and balances to them.
+
+## Complexity and Bootstrap Issues
+
+A new build system, and any build system for that matter, indirectly brings its own dependencies to the Nix project.
+
+Ideally, such dependencies should be minimal, both in number and complexity, with extra good points for dependencies already present (e.g the ubiquitous C compiler).
+
+About this specific point, a non-negligible drawback is: the reference implementation of Meson is written in Python. At least theoretically, it brings the necessity of including a Python 3 interpreter on the _bootstrap route_ of Nix.
+
+However, some points can be laid out on the opposite side:
+
+1. By design, the meson reference evaluator depends only on Python, avoiding the use of extra libraries.
+2. Also by design, the implementation language is never exposed to the meson DSL. Because of it, the reimplementation of Meson in other programming languages becomes way more facilitated.
```

Comment:
```suggestion
2. Also by design, the implementation language is never exposed to the meson DSL. Because of this, the re-implementation of Meson in other programming languages becomes possible
```

Reply:
Reworded